src/lib/comms/sol-lwm2m.c

```diff
@@ -3658,8 +3978,81 @@ register_with_server(struct sol_lwm2m_client *client,
     if (binding)
         sol_blob_unref(binding);
     return r;
+}
 
-#undef ADD_QUERY
+static bool
+bootstrap_request_reply(void *data, struct sol_coap_server *server,
+    struct sol_coap_packet *pkt,
+    const struct sol_network_link_addr *server_addr)
+{
+    struct server_conn_ctx *conn_ctx = data;
+    uint8_t code;
+    int r;
+
+    SOL_BUFFER_DECLARE_STATIC(addr, SOL_NETWORK_INET_ADDR_STR_LEN);
+
+    sol_coap_packet_unref(conn_ctx->pending_pkt);
+    conn_ctx->pending_pkt = NULL;
+
+    if (!pkt && !server_addr) {
+        SOL_WRN("Bootstrap request timeout");
+        SOL_INT_CHECK_GOTO(++conn_ctx->addr_list_idx,
+            == conn_ctx->server_addr_list.len, err_exit);
+        r = bootstrap_with_server(conn_ctx->client, conn_ctx);
+        SOL_INT_CHECK_GOTO(r, < 0, err_exit);
+        SOL_WRN("Trying another address");
+        return false;
+    }
+    if (!sol_network_link_addr_to_str(server_addr, &addr))
+        SOL_WRN("Could not convert the server address to string");
+
+    sol_coap_header_get_code(pkt, &code);
+    SOL_INT_CHECK_GOTO(code, != SOL_COAP_RESPONSE_CODE_CHANGED, err_exit);
+
+    SOL_DBG("Bootstrap process with server %.*s can start",
+        SOL_STR_SLICE_PRINT(sol_buffer_get_slice(&addr)));
+
```

Comment:
Done.

I thought I would be able to do the same inside `register_reply()`, but I got a few errors when trying it. :/